File: api-java-sql/src/main/java/com/nebulahotel/repository/QuartosRepository.java
package com.nebulahotel.repository;

import com.nebulahotel.model.Quartos;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface QuartosRepository extends JpaRepository<Quartos, Integer> {

    List<Quartos> findByStatusIgnoreCase(String status);

    boolean existsByNumero(String numero);

    Optional<Quartos> findByNumero(String numero);
}

File: api-kotlin-nosql/src/main/kotlin/com/nebulahotel/repository/QuartosRepository.kt
package com.nebulahotel.repository

import com.nebulahotel.model.Quartos
import org.springframework.data.mongodb.repository.MongoRepository

interface QuartosRepository : MongoRepository<Quartos, Int> {
    fun findByStatusIgnoreCase(status: String): List<Quartos>
    fun existsByNumero(numero: String): Boolean
    fun findByNumero(numero: String): Quartos?
}
